src/main/java/net/minestom/server/event/player/PlayerMoveEvent.java

```diff
@@ -11,15 +11,27 @@
  */
 public class PlayerMoveEvent extends PlayerEvent implements CancellableEvent {
 
+    private final Position currentPosition;
     private Position newPosition;
 
     private boolean cancelled;
 
     public PlayerMoveEvent(@NotNull Player player, @NotNull Position newPosition) {
         super(player);
+        this.currentPosition = player.getPosition();
         this.newPosition = newPosition;
     }
 
+    /**
+     * Gets current position of the player.
+     *
+     * @return the current position
+     */
+    @NotNull
+    public Position getCurrentPosition() {
```

Comment:
In my opinion this should be `getOldPosition`. It is correct that the position has yet to change, however it can be confusing if somebody does not pay attention.